### PR TITLE
Setup common models before building any model

### PR DESCRIPTION
**Pull request by pramodk**
@ferdonline : See analysis in https://bbpteam.epfl.ch/project/issues/browse/CNEUR-343
If there is better way to fix this, please go ahead.

### CODE REVIEW

**Review by matz-e — APPROVED** [review]
LGTM from the Spack side.

File: var/spack/repos/builtin/packages/neurodamus-model/package.py [hunk 1]
```diff
@@ -59,11 +68,6 @@ def merge_hoc_mod(self, spec, prefix):
         """
         core_prefix = spec['neurodamus-core'].prefix
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py:72]:
```suggestion
```

File: var/spack/repos/builtin/packages/neurodamus-model/package.py [hunk 2]
```diff
@@ -42,7 +42,16 @@ class NeurodamusModel(SimModel):
     # and we must bring their dependencies.
     depends_on('zlib')  # for hdf5
 
-    phases = ['build_model', 'merge_hoc_mod', 'build', 'install']
+    phases = ['setup_common_mods', 'build_model', 'merge_hoc_mod', 'build', 'install']
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py]:
```suggestion
    phases = [
        'setup_common_mods', 'build_model', 'merge_hoc_mod', 'build', 'install'
    ]
```

**Review by matz-e — CHANGES_REQUESTED** [review]
Missed the flake8 bits, though.

**Review by matz-e — DISMISSED** [review]

**Review by ferdonline — DISMISSED** [review]
That makes all the sense! The fix looks good to me!

**Comment by pramodk** [comment]
Please retest

**Comment by pramodk** [comment]
please retest

**Review by matz-e — APPROVED** [review]

**Comment by pramodk** [comment]
@ferdonline @iomaganaris @jorblancoa  : I am merging this long pending change! So if something breaks, you know where to look at :)

**Pull request merged**